packages/codecs-core/src/codec.ts

```diff
@@ -146,3 +142,61 @@ export function createCodec<From, To extends From = From>(
         write: codec.write,
     });
 }
+
+export function isFixedSizeCodec<T>(encoder: Encoder<T>): encoder is FixedSizeEncoder<T>;
+export function isFixedSizeCodec<T>(decoder: Decoder<T>): decoder is FixedSizeDecoder<T>;
+export function isFixedSizeCodec<T, U extends T = T>(codec: Codec<T, U>): codec is FixedSizeCodec<T, U>;
+export function isFixedSizeCodec<T, U extends T = T>(codec: Encoder<T> | Decoder<T> | Codec<T, U>): boolean {
+    return 'fixedSize' in codec && typeof codec.fixedSize === 'number';
+}
+
+export function assertIsFixedSizeCodec<T>(
+    encoder: Encoder<T>,
+    message?: string
+): asserts encoder is FixedSizeEncoder<T>;
+export function assertIsFixedSizeCodec<T>(
+    decoder: Decoder<T>,
+    message?: string
+): asserts decoder is FixedSizeDecoder<T>;
+export function assertIsFixedSizeCodec<T, U extends T = T>(
+    codec: Codec<T, U>,
+    message?: string
+): asserts codec is FixedSizeCodec<T, U>;
+export function assertIsFixedSizeCodec<T, U extends T = T>(
+    codec: Encoder<T> | Decoder<T> | Codec<T, U>,
+    message?: string
+): void {
+    if (!isFixedSizeCodec(codec as Codec<T, U>)) {
+        // TODO: Coded error.
+        throw new Error(message ?? 'Expected a fixed-size codec, got a variable-size one.');
+    }
+}
+
+export function isVariableSizeCodec<T>(encoder: Encoder<T>): encoder is VariableSizeEncoder<T>;
+export function isVariableSizeCodec<T>(decoder: Decoder<T>): decoder is VariableSizeDecoder<T>;
+export function isVariableSizeCodec<T, U extends T = T>(codec: Codec<T, U>): codec is VariableSizeCodec<T, U>;
+export function isVariableSizeCodec<T, U extends T = T>(codec: Encoder<T> | Decoder<T> | Codec<T, U>): boolean {
+    return !('fixedSize' in codec) || codec.fixedSize == null;
```

Comment:
Is this always true?

Reply:
Oh I see. Yeah it typically shouldn't be there at all. But it felt a bit weak to me to only check the inexistence of a property. For instance, say someone has a `VariableSizeCodec` object but they explicitly set `fixedSize` to `undefined` instead of not including it. Then this check would fail.

Alternatively, I could check for the existence of `getSizeFromValue` function instead and ignore the `fixedSize` attribute. What do you think?

Reply:
Actually no that won't work with decoders. I just changed it to use `!isFixedSizeCodec()`. 😅

Reply:
Looks really good! Yeah I was just confused on why a null value there would result in a pass. But this lgtm!